### PR TITLE
libbonoboui: replace dep to pangox-compat with pango 

### PR DESCRIPTION
**Pull request by someuser12**
#### Description

The aim of this four commits is to be able to build and use unison (in its gtk variant) on Monterey machines with M1 CPUs. Therefore, one needs to be able to build the GTK-related dependencies of unison.
By the way, replacing the broken (on Monterey) pangox-compat with pango in the dependencies of libbonoboui is a step on the way of retiring pangox-compat (see https://trac.macports.org/ticket/64228).

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 12.1 21C52 arm64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: gnome/libbonoboui/Portfile
Comment (reneeotten) [inline gnome/libbonoboui/Portfile:47]: when changing a `depends_lib` dependency you'll need to increase the `revision`

Comment (someuser12) [reply gnome/libbonoboui/Portfile:47]: Thanks, that's right, I had forgotten to do it.

Comment (ryandesign) [reply gnome/libbonoboui/Portfile:47]: Any port that depends on pango should do so using `path:lib/pkgconfig/pango.pc:pango` so that pango-devel can satisfy it.
```suggestion
                    path:lib/pkgconfig/pango.pc:pango
```

**Review by ryandesign — CHANGES_REQUESTED** [review]
> replacing the broken (on Monterey) pangox-compat with pango in the dependencies of libbonoboui is a step on the way of retiring pangox-compat

If this works, great; it would mean libbonoboui at some point stopped using pangox-compat. Replacing pangox-compat with pango is of course not a solution for any software that is still using the pangox-compat functionality (which used to be in pango but was removed at some point).

File: lang/mozjs78/Portfile
Comment (ryandesign) [inline lang/mozjs78/Portfile:12]: Has Dave agreed to maintain this port?

File: lang/mozjs78/Portfile
Comment (ryandesign) [inline lang/mozjs78/Portfile:50]: Why is 12.1 the max SDK? What happens when the macOS 12.2 SDK is released?

File: lang/mozjs78/Portfile
Comment (ryandesign) [inline lang/mozjs78/Portfile:51]: Why is `${build_arch}` being printed out here?

File: security/policykit/Portfile
Comment (ryandesign) [inline security/policykit/Portfile:10]: When the version is increased, the revision should be reset to 0.
```suggestion
revision            0
```

File: lang/mozjs78/Portfile
Comment (someuser12) [reply lang/mozjs78/Portfile]: Thanks, this was a leftover of the debugging process. Removed.

File: lang/mozjs78/Portfile
Comment (someuser12) [reply lang/mozjs78/Portfile:12]: No, sorry. This is a mistake due to a copy from the mozjs60 Portfile. What is Dave's opinion about policykit he maintains and about the new mozjs78 port?

File: lang/mozjs78/Portfile
Comment (someuser12) [reply lang/mozjs78/Portfile:50]: Here advice is really welcome. Officially, the max supported version of the SDK is 10.15.4 (if the installed SDK is more recent, the build scripts suggest downgrading to 10.15.4 using e.g. https://github.com/phracker/MacOSX-SDKs), and it turns out that building with the 12.1 SDK also succeeds (on Mac OS 12.1 at least), and build also succeeds under Mac OS 10.15 and 11 in the CI bots. Hence this new max SDK in the Portfile. What would be the best option?

File: net/unison/Portfile
Comment (pmetzger) [inline net/unison/Portfile:6]: I'd suggest we update Unison itself in a distinct step. In particular, the rest of this may take a bit of time to shake out, but Unison could be updated right now with little difficulty.

File: lang/mozjs78/Portfile
Comment (someuser12) [reply lang/mozjs78/Portfile:50]: To be more precise, the build scripts of mozjs 78 suggest downloading an older version of the SDK, possibly in a local directory and, if needed, pointing at this directory thanks to ``https://github.com/phracker/MacOSX-SDKs'' in the mozconfig. If we don't want to build with the newer SDK (12.1 or later) installed system-wide on the machine where the build happens, this might be an option (download the older SDK in a temporary directory, build mozjs78, remove the SDK), but this means relying on a third party from which the binaries of the older SDK are downloaded.

File: net/unison/Portfile
Comment (someuser12) [reply net/unison/Portfile:6]: Do you mind if we also add `arm64` as supported architecture for unison (like ocaml does)?

Comment (pmetzger) [reply net/unison/Portfile:6]: That's fine, but again, one can do this in two pull requests and thus get Unison updated immediately.

Comment (someuser12) [reply net/unison/Portfile:6]: Done in https://github.com/macports/macports-ports/pull/13672. I still need to remove unison update from this PR.

Comment (pmetzger) [reply net/unison/Portfile:6]: I've merged that PR.

File: security/policykit/Portfile
Comment (evanmiller) [inline security/policykit/Portfile:38]: This suggestion is outside your responsibilities, but it would be nice to keep around the old policykit (which relies on Rust-free `mozjs60`) for 10.4/10.5/10.6 users. We do something similar for `librsvg`.

File: lang/mozjs78/Portfile
Comment (evanmiller) [inline lang/mozjs78/Portfile:115]: The `rust` build requirement is going to knock out 10.4 and 10.5. This whole block (and associated patches) can be replaced with a `known_fail yes` and error message.

10.6 is also knocked out for now, but rumor has it that @catap is working on a fix...

**Comment by evanmiller** [comment]
See #13789 for an alternative approach to updating policykit (using `duktape` instead of `mozjs78`). Duktape does not require Rust, and so should build on all supported MacPorts systems.

File: lang/mozjs78/Portfile
Comment (catap) [reply lang/mozjs78/Portfile:115]: @evanmiller yes, I'm almost finished some rust on macOS 10.6 :)

**Comment by pmetzger** [comment]
So what's our status here?

**Comment by evanmiller** [comment]
In my opinion #13789 should be preferred (waiting on maintainer feedback over there), and this PR should be trimmed down to include only the `libbonoboui` commit.

The only other mozjs consumer (`gjs`) is already moving to `mozjs91`, so I think any mozjs efforts should focus on that release and just skip `mozjs78`.

**Comment by someuser12** [comment]
Done. Only the libbonoboui commit is left.

**Comment by pmetzger** [comment]
Okay, so am I correct in assuming this is now ready for commit? I recommend if so given that it's just a commit for libbonoboui that the commit message be updated...

**Review by neverpanic — APPROVED** [review]

**Comment by neverpanic** [comment]
Even usptream seems to suggest retiring pangox-compat: https://gitlab.gnome.org/GNOME/pango/-/issues/644.

Looks good to me, merging.

**Pull request merged**
